src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,624 @@
+        ChronoField.YEAR.checkValidValue (prolepticYear);
+        InternationalFixedChronology.DAY_OF_YEAR_RANGE.checkValidValue (dayOfYear, ChronoField.DAY_OF_YEAR);
+        boolean leap = InternationalFixedChronology.INSTANCE.isLeapYear (prolepticYear);
+
+        if (dayOfYear > (InternationalFixedChronology.DAYS_IN_YEAR + 1) && !leap) {
+            throw new DateTimeException ("Invalid date 'DayOfYear " + dayOfYear + "' as '" + prolepticYear + "' is not a leap year");
+        }
+
+        int month = ((dayOfYear - 1) / InternationalFixedChronology.DAYS_IN_MONTH) + 1;
+
+        // In leap years, the leap-month is shorter than the following month, so needs to be adjusted.
+        if (leap && month == InternationalFixedChronology.MONTHS_IN_YEAR && dayOfYear >= InternationalFixedChronology.DAYS_IN_YEAR - InternationalFixedChronology.DAYS_IN_MONTH + 1) {
+            month++;
+        }
+
+        // Subtract days-at-start-of-month from days in year
+        int dayOfMonth = dayOfYear - (month - 1) * InternationalFixedChronology.DAYS_IN_MONTH;
+
+        // Adjust for shorter inserted leap-month.
+        if (month == InternationalFixedChronology.MONTHS_IN_YEAR + 1) {
+            dayOfMonth += InternationalFixedChronology.DAYS_IN_MONTH;
+        }
+
+        return of (prolepticYear, month, dayOfMonth);
+    }
+
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the epoch-day.
+     *
+     * @param epochDay the epoch day to convert based on 1970-01-01 (ISO)
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the epoch-day is out of range
+     */
+    public static InternationalFixedDate ofEpochDay (final long epochDay) {
+        ChronoField.EPOCH_DAY.range ().checkValidValue (epochDay, ChronoField.EPOCH_DAY);
+        long zeroDay = epochDay + InternationalFixedChronology.DAYS_0000_TO_1970;
+
+        if (zeroDay < 0) {
+            throw new DateTimeException ("Negative epoch invalid: " + epochDay);
+        }
+
+        long yearEst = (400 * zeroDay) / DAYS_PER_CYCLE;
+        long doyEst = zeroDay - (365 * yearEst + yearEst / 4 - yearEst / 100 + yearEst / 400);
+        int month = 1 + (int) ((doyEst - 1 ) / 28);
+        int dom = 1 + (int) ((doyEst - 1) % 28);
+
+        if (month == 14) {
+            month = 13;
+            dom += 28;
+        }
+
+        // check year now we are certain it is correct
+        int year = ChronoField.YEAR.checkValidIntValue (yearEst);
+
+        return new InternationalFixedDate (year, month, dom);
+    }
+
+    private static InternationalFixedDate resolvePreviousValid (final int prolepticYear, final int month, final int day) {
+        int monthR = Math.min (month, InternationalFixedChronology.MONTHS_IN_YEAR);
+        int dayR = Math.min (day, InternationalFixedChronology.DAYS_IN_MONTH + (month == 6 || month == 13 ? 1 : 0));
```

Comment:
The max value for day-of-month isn't taking into account if the year is actually a leap-year.